WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java

```diff
@@ -993,6 +749,32 @@ public void onInputUpdated(String input) {
         dialog.show(getFragmentManager(), null);
     }
 
+    private void showLocationPicker() {
+        if (!isAdded()) {
+            return;
+        }
+        PlacePicker.IntentBuilder builder = new PlacePicker.IntentBuilder();
+        // Pre-pick the previous selected location if any
+        LatLng latLng = null;
+        if (mPostLocation != null) {
+            latLng = new LatLng(mPostLocation.getLatitude(), mPostLocation.getLongitude());
+        } else if (mPost.hasLocation()) {
+            PostLocation location = mPost.getLocation();
+            latLng = new LatLng(location.getLatitude(), location.getLongitude());
+        }
+        if (latLng != null) {
+            builder.setLatLngBounds(new LatLngBounds(latLng, latLng));
+        }
+        // Show the picker
+        try {
+            startActivityForResult(builder.build(getActivity()), ACTIVITY_REQUEST_CODE_PICK_LOCATION);
+        } catch (GooglePlayServicesNotAvailableException nae) {
+            ToastUtils.showToast(getActivity(), R.string.post_settings_error_placepicker_missing_play_services);
+        } catch (GooglePlayServicesRepairableException re) {
+            GooglePlayServicesUtil.getErrorDialog(re.getConnectionStatusCode(), getActivity(), 0);
```

Comment:
Since `GooglePlayServicesUtil.getErrorDialog` is deprecated, can we use the `GoogleApiAvailability.getInstance().getErrorDialog` instead? Also, can we add a result code for this action, we don't want another dev adding a result code of `0` and accidentally handling this case as well, better safe than to be sorry. So, I think it should look like below:
```
GoogleApiAvailability.getInstance().getErrorDialog(getActivity(), re.getConnectionStatusCode(), PLAY_SERVICES_RESOLUTION_REQUEST);
```